### PR TITLE
Add k8s.cluster.name through resource attributes

### PR DESCRIPTION
**Pull request by cedricziel**
The concrete change here is injecting a new transform processor because the attributes processor cannot write to/access the resource.

### CODE REVIEW

File: charts/k8s-monitoring/templates/agent_config/_traces.river.txt [hunk 1]
```diff
@@ -43,22 +43,31 @@ otelcol.processor.batch "trace_batch_processor" {
...
 }
+
 {{- end }}
 
+otelcol.processor.transform "trace_transform_processor" {
+  trace_statements {
+    context = "resource"
+    statements = [
+      // Accessing a map with a key that does not exist will return nil.
+      "set(attributes[\"k8s.cluster.name\"], \"{{ required ".Values.cluster.name is a required value. Please set it and try again." .Values.cluster.name }}\") where attributes[\"k8s.cluster.name\"] == nil"
```
Comment (petewall) [inline charts/k8s-monitoring/templates/agent_config/_traces.river.txt:57]: I think this needs a `,` at the end.

**Review by petewall — COMMENTED** [review]
Can you help me understand what the difference is here?

**Comment by cedricziel** [comment]
> Can you help me understand what the difference is here?

Absolutely @petewall . The attributes processor works on signal level attributes, while the transform processor can also set resource-level attributes. The resource level is one level above signals and the same resource would ideally be associated with concrete metrics, logs and traces.

https://opentelemetry.io/docs/specs/otel/resource/sdk/

The concrete change here is injecting a new transform processor because the attributes processor cannot write to/access the resource.

**Comment by petewall** [comment]
Closing per @mar4uk 's comment.

**Pull request closed without merging**